packages/extension-ui/src/components/View.tsx

```diff
@@ -2,23 +2,38 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import React from 'react';
-import styled from 'styled-components';
+import React, { useState } from 'react';
+import styled, { ThemeProvider } from 'styled-components';
+import { AvailableThemes, chooseTheme, themes, ThemeToggleContext } from '.';
 
 interface Props {
   children: React.ReactNode;
   className?: string;
 }
 
 function View ({ children, className }: Props): React.ReactElement<Props> {
+  const [theme, setTheme] = useState(chooseTheme());
+
+  const toggleTheme = (theme: AvailableThemes): void => {
+    localStorage.setItem('theme', theme);
+    setTheme(theme);
+  };
+
   return (
-    <main className={className}>
-      {children}
-    </main>
+    <ThemeToggleContext.Provider value={{
+      setTheme: toggleTheme,
+      isDark: theme === 'dark'
```

Comment:
Seems a bit unnecessary, as the current theme can be found in StyledComponent's [ThemeContext](https://www.styled-components.com/docs/advanced#via-usecontext-react-hook).

Still, as you don't export the style objects itself, it becomes a bit tricky.

Maybe we can just hold the value of the themeId (`dark`, `light`) as `currentTheme` here and check that in `Settings`?

Reply:
Perhaps a better idea: each theme can have an `id` field. Than we can use `ThemeContext` to get the current one and see if it's id matches the id of the dark theme (`checked={theme.id === themes.dark.id}`)